src/helpers/getLocatorBase.ts

```diff
@@ -188,7 +188,21 @@ export class GetLocatorBase<LocatorSchemaPathType extends string> {
 	): void {
 		const schema = schemasMap.get(locatorSchemaPath);
 		if (schema) {
-			schemasMap.set(locatorSchemaPath, this.deepMerge(schema, updateData));
+			const updatedSchema = this.deepMerge(schema, updateData);
+
+			if (
+				"update" &&
+				"updates" &&
+				"getNestedLocator" &&
+				"getLocator" &&
+				"locatorSchemaPath" &&
+				"locatorMethod" &&
+				"schemasMap" in schema
+			) {
```

Comment:
Ah haha yes, you're correct. Think a seperate method with your suggested soluton for checking is the way to go :) 

Ironically the current and the correct if statement you pointed out effectively give the same result, as "schemasMap" and the rest of the properties are only present in a LocatorSchemaWithMethods object. The schemasMap provided to the methods applyUpdate and applyUpdates will always as a minimum have one key which refrences the LocatorSchemaWithMethods object it belongs to, e.g.:
![bilde](https://github.com/DyHex/POMWright/assets/1068803/0b56b0ea-3813-4acd-92a6-c6a5d5d3c860)
While the remaining key's if any will contain a copy of LocatorSchema that make up each subpath of the LocatorSchemaWithMethods' full locatorSchemaPath. 

Thus the check might not really be needed in the applyUpdate method, since it only uses the complete locatorSchemaPath which should always be a LocatorSchemaWithMethods object, but doesn't hurt to check. 
The check must be performed in the applyUpdates method though, since it can update any of the LocatorSchema in the map that make up the full locatorSchemaPath, including the LocatorSchemaWithMethods object.